src/main/java/io/jenkins/plugins/audit/listeners/BuildListener.java

```diff
@@ -48,9 +49,41 @@ public void onStarted(Run run, TaskListener listener) {
     }
 
     /**
-     * Returns a registered {@link BuildStartListener} instance.
+     * Fired when a build is completed, event logged via Log4j-audit.
+     *
+     * @param run of type Run having the build information
+     * @param listener of type TaskListener that the onCompleted method expects
+     */
+    @Override
+    public void onCompleted(Run run, TaskListener listener) {
+        BuildFinish buildFinish = LogEventFactory.getEvent(BuildFinish.class);
+
+        List causeObjects = run.getCauses();
+        List<String> causes = new ArrayList<>(causeObjects.size());
+        for (Object cause: causeObjects) {
+            Cause c = (Cause)cause;
+            causes.add(c.getShortDescription());
+        }
+
+        buildFinish.setBuildNumber(run.getNumber());
+        buildFinish.setCause(causes);
+        buildFinish.setProjectName(run.getParent().getFullName());
+        buildFinish.setTimestamp(new Date(run.getStartTimeInMillis() + run.getDuration()).toString());
```

Comment:
This math should work, but a more appropriate way to calculate this using `java.time` classes would be something like this:

```
Instant start = Instant.ofEpochMillis(run.getStartTimeInMillis());
Duration duration = Duration.ofMillis(run.getDuration());
Instant finish = start.plus(duration);
```

Or using some convenience methods:

```
Instant start = Instant.ofEpochMillis(run.getStartTimeInMillis());
Instant finish = start.plusMillis(run.getDuration());
```

In general, you should use `Instant` or `LocalDateTime` (or similar `FooDateTime` classes) instead of `Date` or `Calendar` where possible.

Reply:
Thank you very much for helping me in improving. I will add this change after #23 is accepted as this is only another Date that needs to be formatted appropriately, for now. That will help me to use the DateTimeHelper function also here.